src/cephadm/cephadm

```diff
@@ -133,13 +133,13 @@ class ContainerInfo:
         self.version = version
 
     def __eq__(self, other: object) -> bool:
-        if isinstance(other, ContainerInfo):
-            return (self.container_id == other.container_id
-                    and self.image_name == other.image_name
-                    and self.image_id == other.image_id
-                    and self.start == other.start
-                    and self.version == other.version)
-        return False
+        if not isinstance(other, ContainerInfo):
+            return NotImplemented
```

Comment:
I strongly suggest returning false here instead of `NotImplemented`, unless this is a pattern already established in the cephadm code (and if so, please point me at them).  I think raising an exception from a `==` operation violates the principle of least surprise. If the type is incompatible, returning False makes more sense.

Reply:
Returning a `NotImplemenetd` in this case I guess doesn't violate the POLA. It is the recommendation of Python data model:
https://docs.python.org/3/reference/datamodel.html#object.__eq__
https://docs.python.org/3.6/library/constants.html#NotImplemented

Reply:
Thanks. You learn something new every day. :-)